### ISSUE TITLE
Issue of visualization on windows 10

### ISSUE THREAD

**Issue by Musosuke** [ChiSu001/SAT-HMR#1]
Hello,

I have installed SAT-HMR following the instructions provided. However, since the xformers developers have removed the wheel for version `0.0.28.post1`, I installed `xformers==0.0.27.post2` and used CUDA 11.8 instead of CUDA 12.1.

When I run the demo command:

```bash
python main.py --mode infer --cfg demo
```

I receive the following error message:

```
(sathmr) C:\Users\DengHsian\Desktop\SAT-HMR\sat-hmr>python main.py --mode infer --cfg demo
pyrender is not correctly imported.
C:\Users\DengHsian\Desktop\SAT-HMR\sat-hmr\models\encoders\dinov2\layers\swiglu_ffn.py:43: UserWarning: xFormers is available (SwiGLU)
  warnings.warn("xFormers is available (SwiGLU)")
C:\Users\DengHsian\Desktop\SAT-HMR\sat-hmr\models\encoders\dinov2\layers\attention.py:27: UserWarning: xFormers is available (Attention)
  warnings.warn("xFormers is available (Attention)")
C:\Users\DengHsian\Desktop\SAT-HMR\sat-hmr\models\encoders\dinov2\layers\block.py:33: UserWarning: xFormers is available (Block)
  warnings.warn("xFormers is available (Block)")
WARNING: You are using a SMPL model, with only 10 shape coefficients.
Preparing models...
WARNING: You are using a SMPL model, with only 10 shape coefficients.
Loading pretrained weights: ./weights/sat_hmr/sat_644.pth
C:\Users\DengHsian\Desktop\SAT-HMR\sat-hmr\engines\engine.py:87: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  state_dict = torch.load(args.pretrain_path)
Loading inference images from ./demo
Initialization finished.
221949700 trainable parameters(221949700 total).
Using following threshold(s):  [0.3]
Results will be saved at: ./demo_results\thresh_0.3
inference:   0%| 
```

The key error message is:
```
pyrender is not correctly imported.
```

I traced this message to sat-hmr/utils/visualization.py, where importing pyopengl_platform = 'egl' fails.

I found a similar issue in another project: [VIBE Issue #101](https://github.com/mkocabas/VIBE/issues/101). However, the solution mentioned there indicates that there's no egl.dll in python\Lib\site-packages\OpenGL\DLLS, so the EGL library cannot be loaded.

I understand that the default platform for pyrender is PygletPlatform, but using egl allows rendering on the GPU, which is important if I want to run SAT-HMR in real-time.

My main goal is to extract the SMPL parameters. If the inference pipeline does not rely on pyrender, is it possible to disable the rendering part to bypass this issue?

Upon testing, I noticed that the rendering occurs in infer_funcs.py and is only used for generating rendered images. If I don't need the rendered images, can I safely skip this part?

Could you please help me understand whether disabling the rendering will affect the inference results, specifically in terms of loss calculation? Any guidance on resolving the pyrender import issue or safely disabling rendering would be greatly appreciated.

Thank you for your assistance!

**Comment by ChiSu001** [comment]
Hi @Musosuke , thanks for your interest in our work! 

Unfortunately, we haven't tested our code on Windows and it would be better to use a Linux platform. You can try to set ```os.environ["PYOPENGL_PLATFORM"] = "osmesa"``` (as shown [here](https://github.com/ChiSu001/SAT-HMR/blob/main/utils/visualization.py#L7)) and refer to the [Official Installation Guide](https://pyrender.readthedocs.io/en/latest/install/index.html#installation-guide) of pyrender. Most of the time rendering without a GPU wouldn't be too slow.

If you still encounter issues with pyrender and only want to extract the predicted SMPL parameters, feel free to disable this visulization process. Pyrender is only used for visualizing mesh predictions and won't affect any results. The predictions of our model are contained in a dictionary. You can use the keys ```'pred_poses'``` and ```'pred_betas'``` to extract SMPL parameters following [this](https://github.com/ChiSu001/SAT-HMR/blob/main/engines/funcs/infer_funcs.py#L38).

**Comment by Musosuke** [comment]
Thanks!
